partner_stock_risk/models/stock.py

```diff
@@ -0,0 +1,83 @@
+                    "%s" % moves.mapped('partner_id.name'))
+        return super(StockMove, self).action_done()
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    @api.multi
+    def action_confirm(self):
+        if not self.env.context.get('bypass_risk', False):
+            if (self.location_dest_id.usage == 'customer' and
+                    self.partner_id.risk_exception):
+                exception_msg = _("Financial risk exceeded \n")
+                return self.env['partner.risk.exceeded.wiz'].create({
+                    'exception_msg': exception_msg,
+                    'partner_id': self.partner_id.id,
+                    'origin_reference': '%s,%s' % (self._model, self.id),
+                    'continue_method': 'action_confirm',
+                }).action_show()
+        return super(StockPicking, self).action_confirm()
+
+    @api.multi
+    def action_assign(self):
+        if not self.env.context.get('bypass_risk', False) and any(
+                self.mapped('partner_id.risk_exception')):
+            exception_msg = _("Financial risk exceeded \n")
+            params = self.env.context.get('params', {})
+            if 'purchase.order' not in params and 'sale.order' not in params:
+                return self.env['partner.risk.exceeded.wiz'].create({
+                    'exception_msg': exception_msg,
+                    'partner_id': self.mapped('partner_id')[:1].id,
+                    'origin_reference': '%s,%s' % (self._model, self.id),
+                    'continue_method': 'action_assign',
+                }).action_show()
+        return super(StockPicking, self).action_assign()
+
+    @api.multi
+    def force_assign(self):
+        if not self.env.context.get('bypass_risk', False):
+            if (self.location_dest_id.usage == 'customer' and
+                    self.partner_id.risk_exception):
+                exception_msg = _("Financial risk exceeded \n")
+                return self.env['partner.risk.exceeded.wiz'].create({
+                    'exception_msg': exception_msg,
+                    'partner_id': self.partner_id.id,
+                    'origin_reference': '%s,%s' % (self._model, self.id),
+                    'continue_method': 'force_assign',
+                }).action_show()
+        return super(StockPicking, self).force_assign()
+
+    @api.multi
+    def do_new_transfer(self):
+        if not self.env.context.get('bypass_risk', False):
+            if (self.location_dest_id.usage == 'customer' and
+                    self.partner_id.risk_exception):
+                exception_msg = _("Financial risk exceeded \n")
+                return self.env['partner.risk.exceeded.wiz'].create({
+                    'exception_msg': exception_msg,
+                    'partner_id': self.partner_id.id,
+                    'origin_reference': '%s,%s' % (self._model, self.id),
+                    'continue_method': 'do_new_transfer',
```

Comment:
I see you are repeating this code quite often in this model, just changing the `continue_method`. It would be nice to have that in a separate method to DRY.